.github/workflows/enforce-issue-template.yml

```diff
@@ -0,0 +1,35 @@
+name: enforce issue template
+on:
+  issues:
+    types: ['opened']
+jobs:
+  enforce-issue-template:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: "Look for a matching issue template"
+      shell: bash
+      run: |
+
+        # Look for a matching issue template
+        echo "${{ github.event.issue.body }}" > actual
+        for expected in $(ls .github/ISSUE_TEMPLATE/*.md 2> /dev/null); do
+          echo -n "$(basename $expected)? "
+          if diff -rub <(grep '^#' $expected) <(grep '^#' actual) > /dev/null; then
+            echo "👍 💃"
+            exit 0
+          else
+            echo "👎"
+          fi
+        done
+
+        # Failed to find a match! Close the issue.
+        echo "${{ github.token }}" | gh auth login --with-token
```

Comment:
If you have `GITHUB_TOKEN` set, you shouldn't need this step.

Reply:
I tried without this line and failed. Is setting `GITHUB_TOKEN` something I need to do explicitly?

Reply:
I think so. I should normally be available by default but I guess they stopped due to security reasons?

Reply:
I confirmed that GH redacts the token in logs:

https://github.com/chadwhitacre/chadwhitacre-testing-nothing-to-see-here/runs/1704556932?check_suite_focus=true#step:3:23

Reply:
So I'm gonna stick with what I've got.

Reply:
That's fine. Just curious how this works without any login tho: https://github.com/getsentry/action-prepare-release/blob/940c6e63f659f8c73f7d245422baf0a2d0fd2a5f/action.yml#L83

Reply:
Interesting. The [docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow) make it sound like the token is available but not implicitly. I've confirmed that I can't use `gh issue create` without auth'ing first. Not sure what's up in `action-prepare-release`. Curious! 🧐

```diff
diff --git a/.github/workflows/validate-new-issue.yml b/.github/workflows/validate-new-issue.yml
index 3252396..6ec1853 100644
--- a/.github/workflows/validate-new-issue.yml
+++ b/.github/workflows/validate-new-issue.yml
@@ -24,6 +24,8 @@ jobs:
         done
 
         # Failed to find a match! Close the issue.
+        echo "GITHUB_TOKEN: $GITHUB_TOKEN"
+        gh issue create --title="new issue from action" --body="here we go"
         echo "${{ github.token }}" | gh auth login --with-token
         cat << EOF > comment
         {"body": "Sorry, friend. As far as this ol' bot can tell, your issue does not use one of this repo's available issue templates. Please [try again using a template](https://github.com/${{ github.repository }}/issues/new/choose) so that we have the best chance of understanding and addressing your issue. (And if I'm confused, please [let us know](https://github.com/getsentry/.github/issues/new?title=template+enforcer+is+confused&body=${{ github.event.issue.html_url }}). 😬)\n\n----\n\n[![Did you see the memo about this?](https://user-images.githubusercontent.com/134455/104515469-e04a9c80-55c0-11eb-8e15-ffe9c0b8dd7f.gif)](https://www.youtube.com/watch?v=Fy3rjQGc6lA)"}

```

<img width="610" alt="Screen Shot 2021-01-14 at 4 19 18 PM" src="https://user-images.githubusercontent.com/134455/104650598-709de580-5684-11eb-8f8d-5080a990ac02.png">

Reply:
I just installed `gh` locally to do some testing. After running `gh auth login` I find a config file via `gh auth status`:

```
$ gh auth status
github.com
  ✓ Logged in to github.com as chadwhitacre (~/.config/gh/hosts.yml)
  ✓ Git operations for github.com configured to use ssh protocol.
  ✓ Token: *******************
  
$
```

Reply:
My hunch is that something has written this `hosts.yml` file (or otherwise configured authentication for `gh`) by the time `action-prepare-release` calls `gh issue create`.

Reply:
[Looks like](https://github.com/cli/cli/blob/4750e4ae18735375ef91e84fa5c148783743c379/pkg/cmd/api/api.go#L120-L128) a `GITHUB_TOKEN` would work. ¯\\\_(ツ)\_/¯